tokyo-night-gtk: unstable-2023-01-17 -> unstable-2023-05-30

Madouura opened the pull request:
## Description of changes
[tokyo-night-gtk: refactor and alias to tokyo-night-gtk-packages.full](https://github.com/NixOS/nixpkgs/commit/1e62b5b22ef0d7f0db1e316ae3ae57fb1d1a8b87)

Why do we do this? This package has a LOT of files!
The user may not need some of these themes and icons and the ~100k extra files in them.
Why not just use passthru for these? See: #211340

```console
nix-repl> np = (import /home/mado/Documents/Development/nixpkgs { }).pkgsCross.aarch64-multiplatform
nix-repl> np.__splicedPackages.tokyo-night-gtk ? __spliced                                           
true
nix-repl> np.__splicedPackages.tokyo-night-gtk-packages.full ? __spliced
true
nix-repl> np.__splicedPackages.tokyo-night-gtk-packages.icons.dark ? __spliced
true
nix-repl> np.__splicedPackages.tokyo-night-gtk-packages.themes.dark-b ? __spliced
true
```

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Madouura committed:
tokyo-night-gtk: unstable-2023-01-17 -> unstable-2023-05-30

Madouura commented:
Result of `nixpkgs-review pr 262049` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>tokyo-night-gtk-packages.full</li>
    <li>tokyo-night-gtk-packages.full.icon</li>
    <li>tokyo-night-gtk-packages.full.theme</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2816

Artturin commented:
Splicing is not relevant here, the package has no binaries.

I think it would be better to do it like https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/themes/catppuccin/default.nix does with everything by default and then users have to use overrides to get only certain parts, no need to rename the package.

Artturin commented on pkgs/data/themes/tokyo-night-gtk-packages/default.nix:
```diff
@@ -0,0 +1,105 @@
+    tkIcon ? ""
+  , tkTheme ? ""
+  }: stdenvNoCC.mkDerivation (finalAttrs: {
+    pname = "tokyo-night-gtk-packages" + (
```
```suggestion
    pname = "tokyo-night-gtk" + (
```

Madouura replied:
Ended up going for something a bit different.

Madouura committed:
tokyo-night-gtk: refactor and default to tokyo-night-gtk-variants.full

This allows us to specify which icons/themes we want instead of keeping 100k+ files

Artturin commented on pkgs/data/themes/tokyo-night-gtk/default.nix:
```diff
@@ -1,38 +1,29 @@
+  }) (lib.importJSON ./variants.json);
+in packages // {
+  # Not using `symlinkJoin` because it's massively inefficient in this case
+  full = runCommand "${prefix}_full" { preferLocalBuild = true; } ''
```
```suggestion
  full = runCommand "${prefix}_full" { preferLocalBuild = true; propagatedUserEnvPkgs = [ gtk-engine-murrine ]; } ''
```

Madouura commented:
Done.
